src/main/scala/com/databricks/labs/overwatch/validation/DeploymentValidation.scala

```diff
@@ -0,0 +1,582 @@
+      .withColumn("result", explode(col("concat_columns")))
+      .select("deployment_id", "workspace_id", "result")
+
+
+    val validationReport = resultDF.as[RulesValidationReport].collect().par
+    validationReport.tasksupport = new ForkJoinTaskSupport(new ForkJoinPool(parallelism))
+    validationReport.foreach(validation => {
+      var validationMsg = "SUCCESS"
+      val resultFlag: Boolean = validation.result.passed==null || validation.result.passed.toBoolean
+      if (!resultFlag) {
+        validationMsg = s"""Validation failed:  ${validation.result}"""
+      }
+      val validationDetails = s""" ${validation.result.ruleName}"""
+      synchronized {
+        validationStatus.append(DeploymentValidationReport(resultFlag, getSimpleMsg(validation.result.ruleName), validationDetails, Some(validationMsg), Some(validation.workspace_id)))
+      }
+     })
+    validationStatus
+
+  }
+
+  /**
+   * Provides human readable message for each validation.
+   *
+   * @param ruleName
+   * @return
+   */
+  private  def getSimpleMsg(ruleName: String): String = {
+    ruleName match {
+      case "Common_ETLStoragePrefix" => "ETL Storage Prefix should be common across the workspaces."
+      case "Common_ETLDatabase" => "Workspaces should have a common ETL Database Name."
+      case "Common_ConsumerDatabaseName" => "Workspaces should have a common Consumer Database Name."
+      case "Valid_Cloud_providers" => "Cloud provider can be either AWS or Azure."
+      case "NOTNULL_APIURL" => "API URL should not be empty."
+      case "NOTNULL_SecretScope" => "Secrete scope should not be empty."
+      case "NOTNULL_SecretKey_DBPAT" => "PAT key should not be empty."
+      case "Valid_PrimordialDate" => "Primordial Date should in yyyy-MM-dd format(Ex:2022-01-30) and should be less than current date."
+      case "Valid_MaxDays" => "Max Days should be a number."
+      case "APIURL_Connectivity" => "API URL should give some response with provided scope and key."
+      case "Validate_AuditLogPrefix" => "Folder with Primordial date should be present inside AuditLogPrefix."
+      case "Validate_EventHub" => "Consuming data from EventHub."
+      case "Valid_Excluded_Scopes" => "Excluded scope can be audit:sparkEvents:jobs:clusters:clusterEvents:notebooks:pools:accounts."
+      case "Storage_Access" => "ETL_STORAGE_PREFIX should have read,write and create access"
+      case "Validate_Mount" => "Number of mount points in the workspace should not exceed 50"
+    }
+  }
+
+  /**
+   * Performs mandatory validation before each deployment.
+   */
+  private[overwatch] def performMandatoryValidation(configCsvPath: String,parallelism: Int,deploymentId: String):Unit = {
+    logger.log(Level.INFO, "Performing mandatory validation")
+    validateFileExistence(configCsvPath)
+    val configDF = makeDataFrame(configCsvPath,deploymentId)
+    val groupedRuleSet = RuleSet(configDF.toDF(), by = "deployment_id")
```

Comment:
let's rename this to something more specific to what validation is happening here other than just "groupedRuleSet"

Reply:
done